src/tests/suite/daos_obj.c

```diff
@@ -1538,16 +1598,15 @@ punch_simple_internal(void **state, daos_obj_id_t oid)
 	print_message("Enumerate dkeys:\n");
 	memset(&anchor_out, 0, sizeof(anchor_out));
 	memset(buf, 0, 512);
-	enum_num = 2;
 	total_keys = 0;
 	while (enum_num > 0) {
```

Comment:
Hmm, I think enum_num might be 0 even if daos_anchor is not eof. I would suggest you only use daos_anchor_is_eof() to check if enumeration is done.  Same for akey/record enumeration.

Reply:
ok, I will update